triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/membersync/namedgraphs/UpdatesCreator.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2022 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.triplesgenerator.events.consumers.membersync
+package namedgraphs
+
+import cats.MonadThrow
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import io.renku.graph.config.{GitLabUrlLoader, RenkuUrlLoader}
+import io.renku.graph.model.Schemas.schema
+import io.renku.graph.model._
+import io.renku.graph.model.entities.Person
+import io.renku.graph.model.persons.GitLabId
+import io.renku.graph.model.views.RdfResource
+import io.renku.graph.model.views.SparqlValueEncoder.sparqlEncode
+import io.renku.triplesstore.SparqlQuery
+import io.renku.triplesstore.SparqlQuery.Prefixes
+
+private class UpdatesCreator(implicit renkuUrl: RenkuUrl, gitLabApiUrl: GitLabApiUrl) {
+
+  def insertion(projectPath: projects.Path,
+                members:     Set[(GitLabProjectMember, Option[persons.ResourceId])]
+  ): List[SparqlQuery] = {
+    val (queriesForNewPersons, queriesForExistingPersons) =
+      (separateUsersNotYetInKG.pure[List] ap members.toList).separate
+        .bimap(createPersonsAndMemberLinks(projectPath, _), createMemberLinks(projectPath, _))
+    queriesForNewPersons ::: queriesForExistingPersons
+  }
+
+  def removal(path: projects.Path, members: Set[KGProjectMember]): List[SparqlQuery] = {
+    val projectId = projects.ResourceId(path)
+    members
+      .map(member => generateEdge(projectId, member.resourceId))
+      .map { edge =>
+        SparqlQuery
+          .of(
+            name = "unlink project member",
+            Prefixes of schema -> "schema",
+            s"""|DELETE DATA { 
+                |  GRAPH <${GraphClass.Project.id(projectId)}> { $edge }
```

Comment:
Question for curiosity/educating myself: is it possible to delete many edges "at once", in one query? (instead of generating multiple queries?)